### ISSUE TITLE
Per Tenant JWTBearer Options failed with authentication

### ISSUE THREAD

**Issue by KarthikEK20** [Finbuckle/Finbuckle.MultiTenant#827]
I'm trying to use the latest version 7.0.1. of the Finbuckle to resolve multitenant and to authorize per tenant basis. Hence, I need to set JWTBearerOptions that is Authority based on tenant-resolved domain URL. I have followed doc, and some older samples,

https://www.finbuckle.com/MultiTenant/Docs/Options
https://github.com/Finbuckle/Finbuckle.MultiTenant/issues/303

Both have details on how to do it, but if I follow the same, it doesn't work as expected.  The problem I'm facing is, that I can see the JWTBearerOption set per tenant by using the "builder.Services.ConfigurePerTenant" line, but still, the authorisation fails because of authority don't have any value.

Here is the example code I followed.

// Configure Services
            services.AddHttpContextAccessor();
            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
                                     .AddJwtBearer();
                                     
            services.AddMultiTenant<TenantInfo>()
                .WithStore(ServiceLifetime.Scoped)
                .WithRemoteAuthenticationCallbackStrategy()
                .WithHostStrategy("__tenant__");

            services.ConfigurePerTenant<JwtBearerOptions, TenantInfo>((options, tenantInfo) =>
            {
                options.Authority = "https://mydomain.com/";
                options.Audience = "API"; // API Resource Name
            });
            
 // Configure Middleware
            
            app.MapControllers();
            app.UseStaticFiles();

            app.UseRouting();

            app.UseAuthentication();
            app.UseAuthorization();

            app.UseMultiTenant();
 
//Note - I tried to position the UseMultiTenant() middleware before Use Authentication line, still no use.

If I tried without "PerTenantOption", the authorisation works. Below is the service part code.

// Configure Services
            services.AddHttpContextAccessor();
            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
                                     .AddJwtBearer(options =>
                {
                    options.Authority = "https://mydomain.com/";
                    options.Audience = "API"; // API Resource Name
                }););
                                     
            services.AddMultiTenant<TenantInfo>()
                .WithStore(ServiceLifetime.Scoped)
                .WithRemoteAuthenticationCallbackStrategy()
                .WithHostStrategy("__tenant__");

So, changing the JWTBearerOptions using "services.ConfigurePerTenant<JwtBearerOptions, TenantInfo>" is something irrelevant with actual JWTBearerOptions? Could you please explain what I'm missing?

Thanks in advance.

**Comment by aswin-ingenuous** [comment]
I also experienced the same issue. It was resolved by specifying the scheme name during `ConfigurePerTenant<>()`.
```
services.AddAuthentication()
  .AddJwtBearer(); // this line adds `JwtBearerHandler` using `JwtBearerDefaults.AuthenticationScheme` scheme

// important: use the scheme i.e. `JwtBearerDefaults.AuthenticationScheme`
services.ConfigurePerTenant<JwtBearerOptions, TenantInfo>(JwtBearerDefaults.AuthenticationScheme, (options, tenant) =>
```

**Comment by KarthikEK20** [comment]
Thanks, But I tried to add the JwtBearerDefaults.AuthenticationScheme as you suggested, still the authorization fails.

**Comment by AndrewTriesToCode** [comment]
Hi, do you mind posting a link to a repository with a basic project that reproduces the problem?

**Comment by KarthikEK20** [comment]
I got it fixed @AndrewTriesToCode . Actually @aswin-ingenuous suggestion works perfect. This is what I missed. 

Also, Removing the line .WithRemoteAuthenticationCallbackStrategy() hasn't made any impact.

Somehow, when I tried it yesterday missed to keep the middleware in right orders, So I got the same error. But now I tried this fix with fresh application, and everything works as expected. Thank you guys for the valuable solution.
